pkg/upgrade/drain_test.go

```diff
@@ -0,0 +1,114 @@
+package upgrade
+
+import (
+	"context"
+	"testing"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/log"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestUpgraderDrain(t *testing.T) {
+	tests := []struct {
+		name            string
+		kubeclient      *fake.Clientset
+		role            api.AgentPoolProfileRole
+		nodeName        string
+		wantErr         bool
+		expectedActions [][]string
+	}{
+		{
+			name:     "master-empty",
+			role:     api.AgentPoolProfileRoleMaster,
+			nodeName: "master-000000",
+			expectedActions: [][]string{
+				{"get", "nodes"}},
+			wantErr:    false,
+			kubeclient: fake.NewSimpleClientset(),
+		},
+		{
+			name:     "unknown-role",
+			role:     "cant-find-this",
+			nodeName: "master-000000",
+			expectedActions: [][]string{
+				{"get", "nodes"}},
+			wantErr: true,
```

Comment:
Instead of a boolean, please compare the actual errors.

Reply:
will do